worlds/landstalker/Hints.py

```diff
@@ -0,0 +1,127 @@
+        # Jewel hints are composed of 4 'words' shuffled randomly:
+        # - the name of the player whose world contains said jewel
+        # - the color of the jewel
+        # - two random words from the location name
+        words = generate_blurry_location_hint(item.location, multiworld.per_slot_randoms[player])
+        words[0] = words[0].upper()
+        words[1] = words[1].upper()
+        words.append(item.name.split(' ')[0].upper())
+        words.append(multiworld.player_name[item.location.player].upper())
+        multiworld.per_slot_randoms[player].shuffle(words)
+        hint_text += f"{words[0]} {words[1]} {words[2]} {words[3]}\n"
+    return hint_text.rstrip('\n')
+
+
+def generate_random_hints(multiworld: MultiWorld, this_player: int):
+    hints = {}
+    random = multiworld.per_slot_randoms[this_player]
+
+    hint_texts = []
+
+    # Exclude Life Stock from the hints as some of them are considered as progression for Fahl, but isn't really
+    # exciting when hinted
+    excluded_items = ["Life Stock"]
+
+    progression_items = [item for item in multiworld.itempool if item.advancement and
+                         item.name not in excluded_items]
+
+    local_own_progression_items = [item for item in progression_items if item.player == this_player
+                                   and item.location.player == this_player]
+    remote_own_progression_items = [item for item in progression_items if item.player == this_player
+                                    and item.location.player != this_player]
+    local_unowned_progression_items = [item for item in progression_items if item.player != this_player
+                                       and item.location.player == this_player]
+    remote_unowned_progression_items = [item for item in progression_items if item.player != this_player
+                                        and item.location.player != this_player]
+
+    # Hint-type #1: Own progression item in own world
+    for item in local_own_progression_items:
+        region_hint = item.location.parent_region.hint_text
+        hint_texts.append(f"I can sense {item.name} {region_hint}.")
+
+    # Hint-type #2: Remote progression item in own world
+    for item in local_unowned_progression_items:
+        other_player = multiworld.player_name[item.player]
+        own_local_region = item.location.parent_region.hint_text
+        hint_texts.append(f"You might find something useful for {other_player} {own_local_region}. "
+                          f"It is a {item.name}, to be precise.")
+
+    # Hint-type #3: Own progression item in remote location
+    for item in remote_own_progression_items:
+        other_player = multiworld.player_name[item.location.player]
+        if item.location.game == 'Landstalker':
+            region_hint_name = item.location.parent_region.hint_text
+            hint_texts.append(f"If you need {item.name}, tell {other_player} to look {region_hint_name}.")
```

Comment:
This one works fine currently, since the hint_text property for LandstalkerRegion instances contain more than just the name itself.
Here are a few examples :
- a region named "Mir Tower" has a hint text containing "inside Mir Tower"
- a region named "Greenmaze" (labyrinthic woods, basically) has a hint text containing "among the trees"
- etc...

Doing so allow for more flexibility language-wise, at the cost of this slightly weird looking construct inside the code.